mne/viz/ica.py

```diff
@@ -627,30 +633,48 @@ def plot_ica_scores(ica, scores, exclude=None, labels=None, axhline=None,
     exclude = np.unique(exclude)
     if not isinstance(scores[0], (list, np.ndarray)):
         scores = [scores]
-    n_rows = len(scores)
+    n_scores = len(scores)
+
+    if n_cols is None:
+        # prefer more rows.
+        n_rows = int(np.ceil(np.sqrt(n_scores)))
+        n_cols = int(np.ceil(n_scores / float(n_rows)))
+    else:
+        n_cols = min(n_scores, n_cols)
+        n_rows = int(np.ceil(n_scores / float(n_cols)))
+
     if figsize is None:
-        figsize = (6.4, 2.7 * n_rows)
-    fig, axes = plt.subplots(n_rows, figsize=figsize, sharex=True, sharey=True)
+        figsize = (6.4 * n_cols, 2.7 * n_rows)
+    fig, axes = plt.subplots(n_rows, n_cols, figsize=figsize,
+                             sharex=True, sharey=True)
+
     if isinstance(axes, np.ndarray):
         axes = axes.flatten()
     else:
         axes = [axes]
-    axes[0].set_title(title)
+
+    fig.suptitle(title)
 
     if labels == 'ecg':
         labels = [label for label in ica.labels_ if label.startswith('ecg/')]
+        labels.sort(key=lambda l: l.split('/')[1])  # sort by index
```

Comment:
Alternatively (avoids the conditional):
```suggestion
        labels = [label for label in ica.labels_ if label.startswith('ecg')]
        # sort by index, allowing just `ecg`
        labels.sort(key=lambda l: l.split('/')[:2][-1])
```

Reply:
Assuming I need the same logic in both the 'ecg' and 'eog' cases.
This change fails in the case of : 
```
    ica.labels_ = dict()
    ica.labels_['ecg/0/foo'] = 0
    ica.labels_['ecg'] = 0
```
because it would return `['ecg', 'ecg/0/foo']` when we just want `['ecg']`.

The fact that the current code on master didn't work for the last several years and no complained suggests this isn't a heavily used case.

Reply:
Ahh okay, I assumed that we might actually want both `ecg` and `ecg/0/foo`, but sounds like that's not the case